### PR TITLE
Update tidb-roadmap.md since last LTS

### PR DESCRIPTION
**Pull request by samhld**
- still needs security section
  - still needs final review from TLs and certain PMs

### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-roadmap.md [hunk 1]
```diff
@@ -101,69 +74,29 @@ This roadmap brings you what's coming in the future, so you can see the new feat
...
-              failures
-            </i>
-          </li>
-          <br />
-          <li>
-            <b>TiProxy</b>
-            <br /><i>
-              Keep database connections during cluster upgrade and scale
-              in/out, and avoid impact on applications
+              Control over how background tasks can affect foreground traffic. These tasks could be imports, DDL, TTL, auto-analyze, compactions, etc.
```
Comment (Oreoxmt) [inline tidb-roadmap.md:79]:
```suggestion
              Control over how background tasks, such as imports, DDL, TTL, auto-analyze, and compactions, can affect foreground traffic.
```

File: tidb-roadmap.md [hunk 2]
```diff
@@ -175,29 +108,20 @@ This roadmap brings you what's coming in the future, so you can see the new feat
...
+            <b>MySQL 8.0 compatibility</b>
           </li>
           <br />
-          <li>
-            <b>Table level flashback</b>
-            <br /><i>
-              Support traveling a single table to a specific time via SQL
-            </i>
+                    <li>
+            <b>Unified SQL interface for import, backup/restore, and PiTR</b>
```
Comment (Oreoxmt) [inline tidb-roadmap.md:115]:
```suggestion
            <b>Unified SQL interface for import, Backup & Restore, and PITR</b>
```

File: tidb-roadmap.md [hunk 3]
```diff
@@ -231,60 +151,32 @@ This roadmap brings you what's coming in the future, so you can see the new feat
...
             >
+            <br /><i>Ensure a smooth upgrade experience</i>
           </li>
           <br />
           <li>
-            <b>SQL-based data import</b>
+            <b>TiCDC native integrations with big data systems</b>
             <br /><i
-              >Improve user-friendliness through operational enhancements</i
+              >Snowflake, Iceburg, etc.</i
```
Comment (Oreoxmt) [inline tidb-roadmap.md:168]:
```suggestion
              >Such as Snowflake and Iceburg</i
```

**Comment by samhld** [comment]
@Oreoxmt Final security updates for your review.

File: tidb-roadmap.md [hunk 4]
```diff
@@ -57,38 +44,24 @@ This roadmap brings you what's coming in the future, so you can see the new feat
...
-            >
+            <b>Performance optimization framework for all applicable background tasks, like DDL, TTL, and cluster analysis</b><br />
+            <i>This distributes the workload of these operations throughout the cluster, resulting in accelerated performance and reduced resource utilization on individual nodes. This framework already applies to the <code>ADD INDEX</code> operation</i>
           </li>
           <br />
           <li>
-            <b>Disaggregate TiFlash compute and storage (auto-scaling) </b><br />
-            <i>Realize elastic HTAP resource utilization</i>
+            <b>GA of disaggregated storage and compute architecture and S3 shared storage in TiFlash</b><br />
+            <i>Enables more cost-effective and elastic HTAP</i>
```
Comment (Oreoxmt) [inline tidb-roadmap.md:53]:
```suggestion
            <i>Enable more cost-effective and elastic HTAP</i>
```

File: tidb-roadmap.md [hunk 5]
```diff
@@ -373,8 +255,25 @@ This roadmap brings you what's coming in the future, so you can see the new feat
...
+## Previously delivered roadmap items
+
+You might have been waiting on some items from the last version. The following lists some previously delivered features. For more details, refer to the [v7.1.0 release notes](/releases/release-7.1.0.md).
+
+- Foundation of multi-tenancy framework: resource control quotas and scheduling for resource groups
+- TiCDC supports object storage sink, including Amazon S3 and Azure Blob Storage (GA)
+- Fastest online `ADD INDEX` (GA)
+- TiFlash late materialization (GA)
+- TiFlash supports spill to disk (GA)
+- SQL-based data import (GA)
```
Comment (Oreoxmt) [inline tidb-roadmap.md:267]:
```suggestion
```

File: tidb-roadmap.md [hunk 6]
```diff
@@ -175,29 +106,20 @@ This roadmap brings you what's coming in the future, so you can see the new feat
...
-          <br />
         </ul>
       </td>
       <td>
         <ul>
           <li>
-            <b>Materialized views</b>
-            <br /><i>Support pre-calculation to boost query performance</i>
+            <b>Cascades framework for optimizer</b>
+            <br /><i>Improved framework for query optimization, and makes the optimizer more extensible and future-proof</i>
```
Comment (ran-huang) [inline tidb-roadmap.md:121]:
```suggestion
            <br /><i>Improved framework for query optimization, and make the optimizer more extensible and future-proof</i>
```

**Review by ran-huang — APPROVED** [review]
rest LGTM

File: tidb-roadmap.md [hunk 7]
```diff
@@ -101,55 +74,18 @@ This roadmap brings you what's coming in the future, so you can see the new feat
...
-              that share the same cluster
-            </i>
-          </li>
-          <br />
-          <li>
-            <b>TiCDC/PITR recovery objective enhancements</b>
-            <br /><i>
-              Increase business continuity and minimize the impact of system
-              failures
+              Control over how background tasks, such as imports, DDL, TTL, auto-analyze, and compactions, can affect foreground traffic.
```
Comment (Oreoxmt) [inline tidb-roadmap.md:79]:
```suggestion
              Control over how background tasks, such as imports, DDL, TTL, auto-analyze, and compactions, can affect foreground traffic
```

**Pull request merged**